### DISCUSSION TITLE
Authentik OAuth/OpenID "Unexpected end of JSON input" Issue

### DISCUSSION THREAD

**Discussion by ncoschrdr** [outline/outline#2716] [Self Hosting]
Hi!

I'm not sure if this is more an issue with Outline or Authentik, so please close this issue if it doesn't fit.

**Problem**
I'm currently trying to set up Outline with Authentik as the OpenID/OAuth Provider, but I can't get the Authentikation to work. I always run in to the error message on the login screen:
```
"Authentication failed — we were 
unable to sign you in at this time. Please 
try again."
```
I've setup quite a few Services with Authentik as the OpenID/OAuth Provider which work fine. I've tried now all possible settings inside Authentik but I can't get it to work and am out of ideas.

**To Reproduce**
1. Go to outline.MYDOMAIN.de
2. Click on 'Continue with Authentik'
3. Login to Authentik
4. Login fails, Error message appears.

This Error message appears onscreen:
![image](https://user-images.githubusercontent.com/55628659/139557692-2c3e6cd8-6b8e-4e49-9eec-83fa568120a1.png)

This Error message can be seen inside the Outline Container Logs
```
FetchError: invalid json response body at https://authentik.MYDOMAIN.de/application/o/userinfo/ reason: Unexpected end of JSON input


    at /opt/outline/node_modules/node-fetch/lib/index.js:272:32


    at processTicksAndRejections (internal/process/task_queues.js:95:5)


    at async OAuth2Strategy._passportOauth.Strategy.userProfile (/opt/outline/build/server/routes/auth/providers/oidc.js:57:25) {


  type: 'invalid-json'


}
```

**Authentik OAuth2 Provider Settings**
![image](https://user-images.githubusercontent.com/55628659/139558116-8483d68f-47e5-42db-83c5-8aac522cac2b.png)
![image](https://user-images.githubusercontent.com/55628659/139558130-dc7f8f83-d973-4eda-a18c-d43dd3f4ab54.png)

**Outline**
- Install:  self hosted (Docker)
- Version: latest tag in Docker Compose

If you need additional Information, please let me know!

Thanks!

Nico

**Comment by redlewel** [comment]
Are you sure that your user has a first and last name in Authentik?  If these fields are left blank it will not work whatsoever as I believe these are required for "profile" scope.  At least that was my experience when using Keycloak.

Also could you post your `.env` file for outline?

**Reply by ncoschrdr** [reply]
I run Authentik and Outline using Docker, these are the environment variables set in my Docker-Compose file:

```
      - PGSSLMODE=disable #
      - SECRET_KEY=${OUTLINE_SECRET_KEY} #
      - UTILS_SECRET=${OUTLINE_UTILS_SECRET} #
      - REDIS_URL=redis://outline-redis:6379 # 
      - DATABASE_URL=postgres://outline:${OUTLINE_POSTGRES_PASSWORD}@outline-postgres:5432/outline #
      - URL=https://outline.${MAINDOMAIN}
      - AWS_ACCESS_KEY_ID=minio #
      - AWS_REGION=us-east-1 #
      - AWS_SECRET_ACCESS_KEY=${MINIO_ROOT_PASSWORD} #
      - AWS_S3_UPLOAD_BUCKET_URL=https://api.minio.${MAINDOMAIN} #
      - AWS_S3_UPLOAD_BUCKET_NAME=outline #
      - AWS_S3_UPLOAD_MAX_SIZE=26214400 #
      - AWS_S3_FORCE_PATH_STYLE=true #
      - AWS_S3_ACL=private #
      - OIDC_CLIENT_ID=${OUTLINE_AUTH_CLIENT_ID} #
      - OIDC_CLIENT_SECRET=${OUTLINE_AUTH_CLIENT_SECRET} #
      - OIDC_AUTH_URI=${OUTLINE_AUTH_URL}/authorize/ #
      - OIDC_TOKEN_URI=${OUTLINE_AUTH_URL}/token #
      - OIDC_USERINFO_URI=${OUTLINE_AUTH_URL}/userinfo #
      - OIDC_USERNAME_CLAIM=preferred_username
      - OIDC_DISPLAY_NAME=Authentik
      - OIDC_SCOPES="openid profile email"
```

**Reply by gnosticJade** [reply]
>       - OIDC_SCOPES="openid profile email"

Try removing the quotes around the scopes - seems to have fixed it for me, also using Authentik. This exact line functions perfectly for me with Authentik 2021.12.5/Outline 0.61.1. Should look like this. `OIDC_SCOPES=openid email profile`

**Reply by imtomjorge** [reply]
It worked without quotes, thanks for your help! :)

**Reply by ToshY** [reply]
Tried this with both latest Authentik/Outline as well as specified versions from @gnosticJade, but didn't make a difference.

**Reply by saschabrockel** [reply]
Did you solve it @ToshY ? Authentik and Outline are working, but I still got the error log:

```
2025-10-23T09:33:02.768856957Z ERR Error processing task in ValidateSSOAccessTask | error=invalid json response body at https://auth.domain.de/application/o/userinfo/ reason: Unexpected end of JSON input stack=BadRequestError: invalid json response body at https://auth.domain.de/application/o/userinfo/ reason: Unexpected end of JSON input
    at InvalidRequestError (/opt/outline/build/server/errors.js:107:34)
    at OIDCClient.userInfo (/opt/outline/build/server/utils/oauth.js:37:47)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async UserAuthentication.validateAccess (/opt/outline/build/server/models/UserAuthentication.js:76:9)
    at async /opt/outline/build/server/queues/tasks/ValidateSSOAccessTask.js:34:18
    at async Promise.all (index 0)
    at async /opt/outline/build/server/queues/tasks/ValidateSSOAccessTask.js:32:24
    at async /opt/outline/node_modules/sequelize/lib/sequelize.js:507:18
    at async ValidateSSOAccessTask.perform (/opt/outline/build/server/queues/tasks/ValidateSSOAccessTask.js:18:5)
    at async Queue.<anonymous> (/opt/outline/build/server/services/worker.js:129:14) 
```

**Reply by ToshY** [reply]
> Did you solve it @ToshY ? Authentik and Outline are working, but I still got the error log:

My comment is over 3 years old (😅) , but I did some backtracking and turns out this was due to my own misconfiguration back then: https://github.com/goauthentik/authentik/issues/3548

**Comment by Mindzy** [comment]
I had the same error here.  After deleting ALLOWED_DOMAINS in .env and selecting email, openid, profile in Scopes, it finally works. In your second screenshot, you have selected only email scope.

**Reply by Ekion-1** [reply]
I'm having the same issues as described here but my list of scopes for the provider is empty. Do I need to manually create these?

**Reply by Mindzy** [reply]
Yep, I created these by myself.
